activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskVariablesPayloadValidator.java

```diff
@@ -0,0 +1,97 @@
+import java.util.Optional;
+import java.util.Set;
+
+import org.activiti.api.task.model.payloads.CreateTaskVariablePayload;
+import org.activiti.api.task.model.payloads.UpdateTaskVariablePayload;
+import org.activiti.common.util.DateFormatterProvider;
+
+public class TaskVariablesPayloadValidator {
+   
+    private final DateFormatterProvider dateFormatterProvider;
+    private final VariableNameValidator variableNameValidator;
+ 
+    public TaskVariablesPayloadValidator(DateFormatterProvider dateFormatterProvider,
+                                         VariableNameValidator variableNameValidator) {
+        this.dateFormatterProvider = dateFormatterProvider;
+        this.variableNameValidator = variableNameValidator;
+    }
+    
+    private Optional<Date> handleAsDate(String value) {
+        try {
+            return Optional.ofNullable(dateFormatterProvider.parse(value));
+        } catch (DateTimeException e) {
+            // ignore exception and return empty: it's not a date so let's keep initial value
+            return Optional.empty();
+        }
+    }
+    
+    public CreateTaskVariablePayload handleCreateTaskVariablePayload(CreateTaskVariablePayload createTaskVariablePayload) {
+        String name = createTaskVariablePayload.getName();
+
+        if (!variableNameValidator.validate(name)) {
+            throw new IllegalStateException("Variable has not a valid name: " + (name != null ? name : "null" ));
+        }
+                
+        Object value = createTaskVariablePayload.getValue();
+        if (value instanceof String) {
+            handleAsDate((String)value).ifPresent(createTaskVariablePayload::setValue);
+        }
+        return createTaskVariablePayload;
+    }
+    
+    public UpdateTaskVariablePayload handleUpdateTaskVariablePayload(UpdateTaskVariablePayload updateTaskVariablePayload) {
+        String name = updateTaskVariablePayload.getName();
+
+        if (!variableNameValidator.validate(name)) {
+            throw new IllegalStateException("You cannot update a variable with not a valid name: " + (name != null ? name : "null" ));
+        }
+        
+        Object value = updateTaskVariablePayload.getValue();
+        
+        if (value instanceof String) {
+            handleAsDate((String) value).ifPresent(updateTaskVariablePayload::setValue);
+        }
+        return updateTaskVariablePayload;
+    }
+    
+    public Map<String, Object> handlePayloadVariables(Map<String, Object> variables) {
+        if (variables != null) {
+            Set<String> mismatchedVars = variableNameValidator.validateVariables(variables); 
+            
+            if (!mismatchedVars.isEmpty()) {
+                throw new IllegalStateException("Variables have not valid names: " + String.join(", ",
+                                                                                                 mismatchedVars));
+                      
+            }        
+     
+            variables.entrySet()
```

Comment:
can you break it down to separate private methods? it is quite hard to understand what it does